components/disk/component.go

```diff
@@ -184,6 +196,121 @@ func (c *component) Close() error {
 	return nil
 }
 
+// evaluateSuggestedActions determines repair actions based on reboot history and disk failure events.
+// Suggests up to 2 reboots, otherwise suggests hardware inspection.
+//
+// Logic mirrors GPU counts component:
+// - Case 1: First disk failure (no previous reboots) → suggest reboot
+// - Case 2: Disk failure after 1 reboot → suggest second reboot
+// - Case 3: Disk failure after 2+ reboots → suggest hardware inspection
+func evaluateSuggestedActions(cr *checkResult, rebootEvents eventstore.Events, diskFailureEvents eventstore.Events) {
```

Comment:
there are some similar functions in GPud, could you abstract them to implement a generic one?